server: TestCheckRestartSafe_AllowMinimumQuorum_Fail failed

server.TestCheckRestartSafe_AllowMinimumQuorum_Fail [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20469145?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20469145?buildTab=artifacts#/) on master @ [4a5fd67651c9030ecefca830fab85c74daf99e39](https://github.com/cockroachdb/cockroach/commits/4a5fd67651c9030ecefca830fab85c74daf99e39):


```
=== RUN   TestCheckRestartSafe_AllowMinimumQuorum_Fail
    test_log_scope.go:171: test logs captured to: /artifacts/tmp/_tmp/0d18d6babea387f19b431b0a26855c08/logTestCheckRestartSafe_AllowMinimumQuorum_Fail3322878937
    test_log_scope.go:82: use -show-logs to present logs inline
    test_server_shim.go:141: cluster virtualization disabled in global scope due to issue: #76378 (expected label: C-bug)
    api_v2_test.go:453: condition failed to evaluate within 45s: waiting for range 77 to have 5 voters (current: 3)
    panic.go:635: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/0d18d6babea387f19b431b0a26855c08/logTestCheckRestartSafe_AllowMinimumQuorum_Fail3322878937
--- FAIL: TestCheckRestartSafe_AllowMinimumQuorum_Fail (50.30s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/obs-prs @cockroachdb/server
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCheckRestartSafe_AllowMinimumQuorum_Fail.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-54475

--- a/pkg/server/api_v2_test.go
+++ b/pkg/server/api_v2_test.go
@@ -444,7 +444,11 @@ func updateAllReplicaCounts(
 ) {
 	_, err := testCluster.ServerConn(0).Exec("ALTER RANGE default CONFIGURE ZONE USING num_replicas=$1", desiredVoters)
 	require.NoError(t, err)
-	require.NoError(t, testCluster.WaitForFullReplication())
+	if desiredVoters == 5 {
+		require.NoError(t, testCluster.WaitFor5NodeReplication())
+	} else {
+		require.NoError(t, testCluster.WaitForFullReplication())
+	}
 
 	ts0 := testCluster.Server(0)
 	stores := ts0.GetStores().(storeVisitor)

--- a/pkg/testutils/serverutils/test_cluster_shim.go
+++ b/pkg/testutils/serverutils/test_cluster_shim.go
@@ -239,6 +239,11 @@ type TestClusterInterface interface {
 	// have no ranges with replication pending.
 	WaitForFullReplication() error
 
+	// WaitFor5NodeReplication ensures that zone configs are applied and
+	// up-replication is performed with new zone configs. This is the case for 5+
+	// node clusters.
+	WaitFor5NodeReplication() error
+
 	// StartedDefaultTestTenant returns whether this cluster started a
 	// default tenant for testing.
 	StartedDefaultTestTenant() bool

